### PR TITLE
Deploy specific project

### PR DESCRIPTION
**Pull request by george-ayris**
#1 

Filters the projects using the -p or --projects flags

### CODE REVIEW

File: generate_manifest.py [hunk 1]
```diff
@@ -30,40 +30,53 @@
...
+
 if __name__ == '__main__':
     parser = argparse.ArgumentParser()
     parser.add_argument('-v', '--versions', default={}, type=json.loads, 
-                        help="Supply specific versions of projects that you like in the manifest. Supplied as a string dictionary. (Will need to escape quotes)")
+                        help="Supply specific versions of projects that you like in the manifest. "
+                             "Supplied as a string dictionary. (Will need to escape quotes)")
     parser.add_argument('-e', '--environment', default="local", type=str,
                         help="Create a manifest based on an existing deploy to an environment.")
+    parser.add_argument('-p', '--projects', default=None, type=str,
```
Comment (domgreen) [inline generate_manifest.py]: Probably wouldnt do this as a `str` ... look at doing it with `nargs="+"` e.g. https://docs.python.org/3/library/argparse.html#nargs

File: .gitignore [hunk 2]
```diff
@@ -108,3 +108,6 @@ venv.bak/
 
 # mypy
 .mypy_cache/
+
+# Pycharm
```
Comment (domgreen) [inline .gitignore:112]: Not using vim 😢

**Review by domgreen — CHANGES_REQUESTED** [review]
Would prefer not to treat the input as a string. Seems fine otherwise. 

*gets rid of georges comments*

**Comment by penderi** [comment]
TROLL TROLL TROLL

File: .gitignore (hunk 2 above)
Comment (penderi) [reply .gitignore:112]: @george-ayris has the skills, just not the inclination

Comment (george-ayris) [reply .gitignore:112]: Need to get my vim setup at work...

File: generate_manifest.py (hunk 1 above)
Comment (george-ayris) [reply generate_manifest.py]: Good idea, will do so

**Comment by george-ayris** [comment]
Changes, as requested

File: generate_manifest.py [hunk 3]
```diff
@@ -30,40 +30,51 @@
...
     parser.add_argument('-e', '--environment', default="local", type=str,
                         help="Create a manifest based on an existing deploy to an environment.")
+    parser.add_argument('-p', '--projects', nargs='+', default=None, type=str,
+                        help="Supply specific projects, to only deploy those projects. "
+                             "Supplied with spaces between project names.")
 
     args = parser.parse_args()
     specific_versions = args.versions
     env = args.environment
+    specific_projects = args.projects
```
Comment (domgreen) [inline generate_manifest.py:51]: Could inline this now your not splitting it. But not needed.

**Review by domgreen — APPROVED** [review]
LGTM

**Pull request merged**